psrp/_compat.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, Jordan Borean (@jborean93) <jborean93@gmail.com>
+# MIT License (see LICENSE or https://opensource.org/licenses/MIT)
+
+import asyncio
+import contextlib
+import functools
+import inspect
+import typing
+
+
+# TODO: Remove once Python 3.6 is dropped.
+if hasattr(contextlib, "asynccontextmanager"):
+    asynccontextmanager = contextlib.asynccontextmanager
+
+else:
+    from async_generator import asynccontextmanager
+
+
+def asyncio_create_task(
```

Comment:
if you're planing on using the httpx anyio backend it would be better to use structured concurrency patterns

```python
async with anyio.create_task_group() as tg:
    ...
```

for background tasks even when just using anyio